packages/expo-updates/android/src/main/java/expo/modules/updates/errorrecovery/ErrorRecovery.kt

```diff
@@ -0,0 +1,103 @@
+package expo.modules.updates.errorrecovery
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import com.facebook.react.ReactInstanceManager
+import com.facebook.react.bridge.DefaultNativeModuleCallExceptionHandler
+import com.facebook.react.bridge.ReactMarker
+import com.facebook.react.bridge.ReactMarkerConstants
+import com.facebook.react.devsupport.DisabledDevSupportManager
+import java.lang.ref.WeakReference
+import kotlin.Exception
+
+class ErrorRecovery(
+  delegate: ErrorRecoveryDelegate
+) {
+  internal val handlerThread = HandlerThread("expo-updates-error-recovery")
+  init {
+    handlerThread.start()
```

Comment:
I've added a separate `initialize` method that can be called earlier than `startMonitoring`, just in case a message is sent (i.e. `notifyNewRemoteLoadStatus`) before the react instance manager is created.